### PR TITLE
[Test] Proxy contract tests

### PR DESCRIPTION
**Pull request by annieke**
## Description

- Initialized a new testing package for @ben-chain's sequencer RPC test and my incoming [chainlink issue](https://github.com/ethereum-optimism/go-ethereum/issues/252) test. Currently just adds a new testing package and ensures it runs when we `$ yarn test` from the base repo. **It does NOT yet add the GitHub workflow.**
- Wrote integration test matching [Chainlink's example](https://github.com/krebernisak/optimism-tutorial/blob/main/test-proxy-events.js)

## Questions
- 
-
-

### Fixes
- Fixes # [Link to Issue]

## Contributing Agreement

- [ ] I have read and understood the [Optimism Contributing Guide and Code of Conduct](https://github.com/ethereum-optimism/optimism-monorepo/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

### CODE REVIEW

**Comment by annieke** [comment]
failing lint for now since it's an empty test, but should be usable.

**Comment by snario** [comment]
Random thought but for some of this Solidity code, is it strictly necessary that we directly add it to the source code or do you think it might be possible that we could pull it from somewhere else? Maybe the `ERC20.sol` or `Proxy.sol` could be pulled from some generic libraries that OpenZeppelin provides for example.

File: README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ There should be a `yarn` script for each package found in `packages`.
 ```bash
 $ yarn test:x-domain
 $ yarn test:tx-ingestion
+$ yarn test:sequencer-interactions
```
Comment (snario) [inline README.md:34]: Since this list is starting to get longer, maybe we should put the content of these commands into a `test.js` script and allow a list of package names to be specified so we could do something like `yarn test x-domain tx-ingestion`.

File: packages/x-domain/package.json [hunk 2]
```diff
@@ -39,6 +39,7 @@
     "@eth-optimism/plugins": "^0.0.16",
     "@eth-optimism/provider": "^0.0.1-alpha.14",
     "@eth-optimism/rollup-core": "^0.0.1-alpha.28",
+    "@eth-optimism/solc": "0.7.6-alpha.1",
```
Comment (snario) [inline packages/x-domain/package.json]: Why did you need to add this here?

File: packages/sequencer-interactions/test/read-proxy-event.spec.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    const recipient = '0x0000000000000000000000000000000000000000';
+    await _transfer(recipient, '1');
+    await _transfer(recipient, '1');
+
+    const _queryFilterTransfer = async (queryContract, filterContract) => {
+      // Get the filter
+      const filter = filterContract.filters.Transfer(null, null, null);
+      // Query the filter
+      return queryContract.queryFilter(filter, 0, 'latest');
+    };
```
Comment (snario) [inline packages/sequencer-interactions/test/read-proxy-event.spec.ts]: Why did you choose to scan the all of the blocks to find the event corresponding to the transfer? As I understand it, you could use the return value of `transferTx.wait()` above to get a [`TransactionReceipt`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt) object which has event logs.

File: packages/x-domain/package.json (hunk 2 above)
Comment (snario) [reply packages/x-domain/package.json]: @annieke from Slack:
> also had to upgrad solc to get their example contracts to compile but now it's breaking our merge checks

Comment (annieke) [reply packages/x-domain/package.json]: originally i thought this was the only package that compiled solidity so i was working from it and later noticed it was from the base integration test `package.json`. i'll remove this, should i also remove it from tx-ingestion @tynes ?

File: packages/sequencer-interactions/test/read-proxy-event.spec.ts (hunk 3 above)
Comment (annieke) [reply packages/sequencer-interactions/test/read-proxy-event.spec.ts]: ^just a copy pasta from https://github.com/krebernisak/optimism-tutorial/blob/main/test-proxy-events.js for now to replicate the error and test that this solves it; down to refactor this after we confirm expected behavior from the sequencer

**Comment by tynes** [comment]
> Random thought but for some of this Solidity code, is it strictly necessary that we directly add it to the source code or do you think it might be possible that we could pull it from somewhere else? Maybe the ERC20.sol or Proxy.sol could be pulled from some generic libraries that OpenZeppelin provides for example.

I agree that its better to import the contracts from common repos

File: packages/sequencer-interactions/test/read-proxy-event.spec.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      upgradeableProxyJson.bytecode,
+      l2Wallet
+    )
+    const proxy = await proxyFactory.deploy(impl.address, Buffer.from(''))
+    await proxy.deployTransaction.wait()
+
+    const contract = new Contract(proxy.address, erc20Json.abi, l2Wallet)
+
+    const _transfer = async (to, amount) => {
+      const transferTx = await contract.transfer(to, amount);
```
Comment (annieke) [inline packages/sequencer-interactions/test/read-proxy-event.spec.ts]: @tynes it builds locally for me now but still running into `CALL_EXCEPTION` when i call transfer here 😞

Comment (annieke) [reply packages/sequencer-interactions/test/read-proxy-event.spec.ts]: could be any of the 4 `transfer`s we call in the test, but still getting this error

**Comment by tynes** [comment]
Hey @annieke still running into problems with this PR?

**Comment by annieke** [comment]
the failing test is an expected behavior so think that means the workflow works!

**Comment by tynes** [comment]
The other tests relying on the ERC20 are now failing

File: yarn.lock [hunk 5]
```diff
@@ -2935,17 +2935,15 @@
   resolved "https://registry.yarnpkg.com/@types/node/-/node-12.20.4.tgz#73687043dd00fcb6962c60fbf499553a24d6bdf2"
   integrity sha512-xRCgeE0Q4pT5UZ189TJ3SpYuX/QGl6QIAOAIeDSbAVAd2gX1NxSZup4jNVK7cxIeP8KDSbJgcckun495isP1jQ==
 
-<<<<<<< HEAD
```
Comment (tynes) [inline yarn.lock]: The `yarn.lock` has merge artifacts in it

File: yarn.lock [hunk 6]
```diff
@@ -11299,11 +11326,7 @@ prepend-http@^2.0.0:
   resolved "https://registry.yarnpkg.com/prepend-http/-/prepend-http-2.0.0.tgz#e92434bfa5ea8c19f41cdfd401d741a3c819d897"
   integrity sha1-6SQ0v6XqjBn0HN/UAddBo8gZ2Jc=
 
-<<<<<<< HEAD
```
Comment (tynes) [inline yarn.lock]: This needs to be deleted

**Comment by tynes** [comment]
I think the easiest way to handle the `yarn.lock` problem would be to delete it and then regenerate it

**Comment by smartcontracts** [comment]
Weird, I thought I did that.

**Comment by snario** [comment]
> I think the easiest way to handle the `yarn.lock` problem would be to delete it and then regenerate it

If you run `yarn` it actually automatically handles merge conflicts

**Comment by snario** [comment]
> Weird, I thought I did that.

I think @tynes reviewed the commit that the notification on Github linked him to and at the same time @smartcontracts pushed code that fixed it?

**Comment by smartcontracts** [comment]
Test failure is expected until https://github.com/ethereum-optimism/go-ethereum/pull/274 is merged.

**Comment by annieke** [comment]
thank you so much @smartcontracts  !!! lmk when u need the +1

**Comment by tynes** [comment]
> Test failure is expected until [ethereum-optimism/go-ethereum#274](https://github.com/ethereum-optimism/go-ethereum/pull/274) is merged.

Passing
https://github.com/ethereum-optimism/optimism-integration/pull/104

**Comment by smartcontracts** [comment]
Can this be approved?
